Sources/NIOCore/AsyncChannel/AsyncChannelInboundStream.swift

```diff
@@ -21,8 +21,57 @@ public struct NIOAsyncChannelInboundStream<Inbound: Sendable>: Sendable {
     @usableFromInline
     typealias Producer = NIOThrowingAsyncSequenceProducer<Inbound, Error, NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark, NIOAsyncChannelInboundStreamChannelHandlerProducerDelegate>
 
+    /// A source used for driving a ``NIOAsyncChannelInboundStream`` during tests.
+    public struct TestSource {
+        @usableFromInline
+        internal let continuation: AsyncStream<Inbound>.Continuation
+
+        @inlinable
+        init(continuation: AsyncStream<Inbound>.Continuation) {
+            self.continuation = continuation
+        }
+
+        /// Yields the element to the inbound stream.
+        ///
+        /// - Parameter element: The element to yield to the inbound stream.
+        @inlinable
+        public func yield(_ element: Inbound) {
+            self.continuation.yield(element)
+        }
+
+        /// Finished the inbound stream.
+        @inlinable
+        public func finish() {
+            self.continuation.finish()
+        }
+    }
+
+    @usableFromInline
+    enum _Backing: Sendable {
+        case asyncStream(AsyncStream<Inbound>)
+        case producer(Producer)
+    }
+
     /// The underlying async sequence.
-    @usableFromInline let _producer: Producer
+    @usableFromInline
+    let _backing: _Backing
+
+    /// Creates a new stream with a source for testing.
+    ///
+    /// This is useful for writing unit tests where you want to drive a ``NIOAsyncChannelInboundStream``.
+    ///
+    /// - Returns: A tuple containing the input stream and a test source to drive it.
+    public static func makeTestingStream() -> (Self, TestSource) {
```

Comment:
Should this be `@inlinable`?

Reply:
It doesn't really matter since this is only for testing purposes.

Reply:
Yeah okay that's fair; the question came from the fact that `makeTestingWriter()` *is* `@inlinable` so it rang the inconsistency bell :P but maybe it's unnecessary on the writer then. I'm aware these are nits though :D